src/Particular.Analyzers/DiagnosticIds.cs

```diff
@@ -23,5 +23,7 @@ public static class DiagnosticIds
         public const string ImproperTryCatchSystemException = "PS0019";
         public const string ImproperTryCatchOperationCanceled = "PS0020";
         public const string MultipleCancellationTokensInATry = "PS0021";
+        public const string DateTimeAssignedToDateTimeOffset = "PS0022";
```

Comment:
Related to my other comment about the name of the analyzer:

```suggestion
        public const string ImplictCastFromDateTimeToDateTimeOffset = "PS0022";
```

I went with "implicit cast" first so that any other implicit case diagnostics sit together in Intellisense, etc.